backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt

```diff
@@ -240,65 +241,73 @@ private fun TextFieldBox(
-                        }
-                        .testTag("textFieldClearButton"),
-                )
+@Composable
+private fun RowScope.TrailingIcon(
+    trailingIcon: BpkIcon?,
+    status: BpkFieldStatus,
+    clearAction: BpkClearAction?,
+    value: TextFieldValue,
+) {
+    var lastIcon by remember { mutableStateOf<Icon?>(null) }
+    val currentIcon = if (trailingIcon != null) {
+        Icon(
+            icon = trailingIcon,
+            contentDescription = null,
+            color = animateColorAsState(
+                when (status) {
+                    is BpkFieldStatus.Disabled -> BpkTheme.colors.textDisabled
+                    else -> BpkTheme.colors.textPrimary
+                },
+                label = "Trailing icon color",
+            ).value,
+        )
+    } else {
+        when {
+            status is BpkFieldStatus.Validated -> Icon(
+                icon = BpkIcon.TickCircle,
+                color = BpkTheme.colors.statusSuccessSpot,
+            )
 
-                else -> Unit
-            }
+            status is BpkFieldStatus.Error -> Icon(
+                icon = BpkIcon.ExclamationCircle,
+                color = BpkTheme.colors.statusDangerSpot,
+            )
 
-            val isAnimationVisible =
-                status is BpkFieldStatus.Validated || status is BpkFieldStatus.Error ||
-                    (status is BpkFieldStatus.Default && clearAction != null && value.text.isNotEmpty())
-            AnimatedVisibility(
-                visible = isAnimationVisible,
-                enter = fadeIn() + scaleIn(),
-                exit = scaleOut() + fadeOut(),
-            ) {
-                lastIcon?.let { it ->
-                    Crossfade(it, label = "textFieldTrailingIcon") {
-                        BpkIcon(
-                            icon = it.icon,
-                            contentDescription = it.contentDescription,
-                            size = it.size,
-                            tint = it.color,
-                            modifier = it.modifier,
-                        )
+            status is BpkFieldStatus.Default && clearAction != null && value.text.isNotEmpty() -> Icon(
+                icon = BpkIcon.CloseCircle,
+                contentDescription = clearAction.contentDescription,
+                color = BpkTheme.colors.textSecondary,
+                modifier = Modifier
+                    .clickable(bounded = false, role = Role.Button) {
+                        clearAction.onClick()
                     }
-                }
-            }
+                    .testTag("textFieldClearButton"),
+            )
+
+            else -> null
+        }
+    }
+    if (currentIcon != null) {
+        lastIcon = currentIcon
+    }
+    AnimatedVisibility(
+        visible = currentIcon != null,
+        enter = fadeIn() + scaleIn(),
+        exit = scaleOut() + fadeOut(),
+    ) {
+        Crossfade(lastIcon!!.icon, label = "textFieldTrailingIcon") {
```

Comment:
only use the icon for the crossfade as otherwise there's some flashing issues when any of the other values change